File: suites/osprey-compute-kernels/reference/cases/digitsum/digitsum.c
/* Sum of decimal digit-sums over 1..N — integer division (n/10) and modulo in recursion. */
#include <stdio.h>
#include <stdint.h>

static int64_t digsum(int64_t n) {
    return n < 10 ? n : (n % 10) + digsum(n / 10);
}

int main(void) {
    int64_t acc = 0;
    for (int64_t i = 1; i < 2000001; i++) acc += digsum(i);
    printf("%lld\n", (long long)acc);
    return 0;
}

File: suites/osprey-compute-kernels/reference/cases/digitsum/digitsum.rs
// Sum of decimal digit-sums over 1..N — integer division (n/10) and modulo in recursion.
fn digsum(n: i64) -> i64 {
    if n < 10 {
        n
    } else {
        n % 10 + digsum(n / 10)
    }
}

fn main() {
    let mut acc: i64 = 0;
    let mut i: i64 = 1;
    while i < 2000001 {
        acc += digsum(i);
        i += 1;
    }
    println!("{}", acc);
}
